frontend/src/pages/ResetPasswordPage.tsx

```diff
@@ -0,0 +1,153 @@
+import { useState, FormEvent } from 'react';
+import { useDispatch } from 'react-redux';
+import { Trans, useTranslation } from 'react-i18next';
+import { Alert } from 'react-bootstrap';
+import { Link, useHistory, useParams } from 'react-router-dom';
+import classNames from 'classnames';
+import { userActions } from '../redux/user';
+import { ModalContent } from '../components/modals/modalparts/ModalContent';
+import { ModalHeader } from '../components/modals/modalparts/ModalHeader';
+import { Input, errorState } from '../components/forms/FormField';
+import { Footer } from '../components/Footer';
+import { paths } from '../routers/paths';
+import { StoreDispatchType } from '../redux';
+import { ReactComponent as CheckIcon } from '../icons/check_icon.svg';
+import css from './ResetPasswordPage.module.scss';
+
+const classes = classNames.bind(css);
+
+export const ResetPasswordPage = () => {
+  const dispatch = useDispatch<StoreDispatchType>();
+  const history = useHistory();
+  const { t } = useTranslation();
+  const { token } = useParams<{
+    token: string | undefined;
+  }>();
+
+  const [done, setDone] = useState(false);
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const confirmationError = {
+    err: errorState(useState('')),
+  };
+
+  const handleReset = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    e.stopPropagation();
+    if (password !== confirmPassword) {
+      confirmationError.err.setMessage(t('Password confirmation error'));
+      return;
+    }
+
+    if (!token) return;
+
+    dispatch(userActions.resetPassword({ token, password })).then((res) => {
+      if (userActions.resetPassword.rejected.match(res)) {
+        if (res.payload?.response?.status === 403)
+          setErrorMessage(t('Token expired'));
+        else if (res.payload?.response?.status === 404)
+          setErrorMessage(t('Token not found'));
+        else if (res.payload) {
+          setErrorMessage(t('Error message', { error: res.payload.message }));
+        }
+      } else {
+        setDone(true);
+      }
+    });
```

Comment:
The nesting here could be reduced (generally makes code easier to understand). Inverting an "if" statement is a great way achieve it. [Example](https://www.refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html)
```suggestion
    const res = await dispatch(userActions.resetPassword({ token, password }));
    if (!userActions.resetPassword.rejected.match(res)) {
      setDone(true);
      return;
    }
    if (res.payload?.response?.status === 403)
      setErrorMessage(t('Token expired'));
    else if (res.payload?.response?.status === 404)
      setErrorMessage(t('Token not found'));
    else if (res.payload)
      setErrorMessage(t('Error message', { error: res.payload.message }));
```

Reply:
Good point, your version looks a lot cleaner when there's no 3 nested if's 👍 Added the suggestion